jsx/App/Stories/Story/Display/TextFormatButton.jsx

```diff
@@ -0,0 +1,243 @@
+
+/* Models a text format button. */
+export function TextFormatButton({ sentences, metadata }) {
+
+    /* 
+    Calls on individual helper functions to gather the morphemes,
+    gloss, translation, and title of the selected sentence. 
+    */
+    function processSentences() {
+        let dependents;
+        if (! sentences[0]) {
+            dependents = sentences["dependents"]
+        } else {
+            dependents = sentences[0]["dependents"];
+        }
+
+        const aingaeWordList = dependents[0]["values"];
+        const morphemeList = dependents[1]["values"];
+        const glossList = dependents[3]["values"];
```

Comment:
A lot of LingView's value comes from its flexibility in supporting arbitrary combinations of tiers. This code assumes that the vernacular words are in dependent tier 0, morphemes are 1, and glosses are 3. That assumption can be wrong. 

For example it would be reasonable for an ELAN file to have the following 8 tiers, in order: 
- narrow transcription
- broad transcription
- morphemes
- morpheme translation into Spanish
- morpheme translation into English
- free translation into Spanish
- free translation into English
- suprasegmental features (e.g. creak, falsetto)

Content curators can set the tier order in ELAN files to be anything they like. FLEx files are more consistent than ELAN, but I expect variation there too. 

In a perfect world, LingView would automatically generate good LaTeX glosses regardless of what tiers are present - but in practice, I don't see a way to do this 100% of the time. So, maybe we need user input. When someone clicks the format button, we can ask them to select which tiers to use. Or, LingView can guess, but we let the user change tiers afterwards when they're viewing the LaTeX format. 

We can consider leaving the assumptions built-in, with the option to add flexibility later. If we do this, I'd like to a) make sure we're using the right format for our corpora, and b) put a note on the output page encouraging users to swap out the tiers if needed.